src/Symfony/Component/Translation/Tests/Writer/TranslationWriterTest.php

```diff
@@ -30,18 +31,43 @@ public function testWriteTranslations()
 
     public function testDisableBackup()
     {
-        $dumper = $this->getMock('Symfony\Component\Translation\Dumper\DumperInterface');
-        $dumper
-            ->expects($this->never())
-            ->method('setBackup');
-        $phpDumper = $this->getMock('Symfony\Component\Translation\Dumper\PhpFileDumper');
-        $phpDumper
-            ->expects($this->once())
-            ->method('setBackup');
+        $nonBackupDumper = new NonBackupDumper();
+        $backupDumper = new BackupDumper();
 
         $writer = new TranslationWriter();
-        $writer->addDumper('test', $dumper);
-        $writer->addDumper('php', $phpDumper);
+        $writer->addDumper('non_backup', $nonBackupDumper);
+        $writer->addDumper('backup', $backupDumper);
         $writer->disableBackup();
+
+        $this->assertTrue($nonBackupDumper->backup, 'backup cannot be disabled if setBackup() method does not exist');
+        $this->assertFalse($backupDumper->backup, 'backup can be disabled if setBackup() method does exist');
+    }
+}
+
+class NonBackupDumper implements DumperInterface
+{
+    public $backup = true;
+
+    public function dump(MessageCatalogue $messages, $options = array())
+    {
+    }
+
+    public function backup($backup)
```

Comment:
You are right. In case the `setBackup()` method is not present we cannot do anything else than making sure that the code does not break by trying to call a non-existent method.